File: hello.js
// console.log("Hello!");

// console.log(typeof 42);
// console.log(typeof 42.5);
// console.log(typeof true);
// console.log(typeof "hello");

// naming convention  of JS is lowerCamelCase
// method's () are mandatory
// console.log((42).toString());
// console.log(typeof (42).toString());
// console.log(parseInt("42", 10));
// console.log(parseFloat("42.5", 10));
// console.log(parseInt("101", 10));
// console.log(parseInt("101", 2));

// const student = "Valerie";
// let age = 21;

// console.log(`${student} is ${age} years-old.`);
// age += 1;
// student = "Super Valerie";
// console.log(`${student} is ${age} years-old next year.`);
// console.log(student + " is " + (age + 1) + " years-old next year.");

const student = "vAlerie";

console.log(student[0]);
console.log(student.length);
// size doesn't work!
console.log(student.slice(1)); // everything from index 1
console.log(student.slice(1, 4)); // everything from index 1 to index 4 excluded

console.log(student.toLowerCase());
console.log(student.toUpperCase());
// no capitalize method in JS!

const students =  ['Curtis', 'Kiyoe', 'Joe'];

const studentsString = students.join()
// JOIN adds commas in JS!!!
console.log(studentsString);
console.log(students.join('~'));

console.log(studentsString.split());
console.log(studentsString.split(''));
console.log(studentsString.split('-'));

console.log(`Hello batch
 1188`);

File: hello.rb
# p "Hello!"

# p 42.class
# p 42.5.class
# p true.class
# p "hello".class

# naming convention of Ruby is lower_snake_case
# methods () are optional
# p 42.to_s
# p "42".to_i

# student = "Valerie"
# age = 21

# p "#{student} is #{age} years-old."
# age += 1
# p "#{student} is #{age} years-old next year."

student = "vAlerie"
# p student.first DOESN'T WORK, only for arrays
p student[0]
p student.length
p student.size
# p student.count DOESN'T WORK, only for arrays
p student[0..2] # from index 0 to index 2
p student[..2] # from start to index 2
p student[1..2] # from index 1 to index 2
p student[1..] # from index 1 to the end
p student[1..-1] # from index 1 to the last
p student[1...-1] # from index 1 to the last excluded
p student.slice(1) # only index 1
p student.slice(1, 3) # from index 1, 3 characters

p student.downcase
p student.upcase
p student.capitalize

students =  ['Curtis', 'Kiyoe', 'Joe'];

students_string = students.join
p students_string

students_string = students.join('-')
p students_string

p students_string.split
p students_string.split('')
p students_string.split('-')